renku_notebooks/api/classes/data_service.py

```diff
@@ -53,26 +62,44 @@ def get_storage_by_id(self, user: User, project_id: int, storage_id: str) -> Clo
             source_path=storage["source_path"],
             target_path=storage["target_path"],
             readonly=storage.get("readonly", True),
+            name=storage["name"],
         )
 
-    def validate_storage_configuration(self, configuration: dict[str, Any]) -> None:
+    def validate_storage_configuration(
+        self, configuration: dict[str, Any], source_path: str
+    ) -> None:
         res = requests.post(self.storage_url + "/storage_schema/validate", json=configuration)
         if res.status_code == 422:
             raise InvalidCloudStorageConfiguration(
                 message=f"The provided cloud storage configuration isn't valid: {res.json()}",
             )
-        if res.status_code != 204:
+        if res.status_code != 204 and res.status_code != 501:
+            # 501 means connection testing isn't supported for the specified provider but the
+            # schema passed validation.
```

Comment:
I thought that we were going to have a separate endpoint for checking the config (validation) and another to check if a connection with the validated config is possible. From this it does not look like that is the case.